### ISSUE TITLE
Choose whether to use aired order or DVD order

### ISSUE THREAD

**Issue by trolley** [tvrenamer/tvrenamer#261]
Some TV series have different aired order than DVD order (Firefly for instance). It would be nice to be able to choose whether you want the file names to come from the aired order or DVD order entries on The TVDB.

**Comment by trolley** [comment]
Sorry, I didn't search first and see there is already an open issue #169

**Comment by eprenamer** [comment]
@trolley Is it something you are able to work on?

**Comment by trolley** [comment]
@eprenamer I can take a look. AFAIK the official TVDB API doesn't support DVD ordering. There's another project I use on Linux though, dbr/tvnamer, that does it but they use their own Python API dbr/tvdb_api.

**Comment by vipuldelwadia** [comment]
TheTVDB has both "aired episode number" and "dvd episode number" (https://api.thetvdb.com/swagger#!/Series/get_series_id_episodes), and we capture this information when we query the API in TVRenamer. The issue is exposing this preference in the UI so that the user can select which ordering they want, probably on a show-by-show basis.

**Comment by trolley** [comment]
@vipuldelwadia Thank you for pointing that out. I asked for DVD ordering support on another project I use and was told it's not available. I guess I should have looked myself :)

**Comment by trolley** [comment]
I finally got a chance to look at this but see that tvrenamer still uses the TVDB API v1 which doesn't include the DVD order information. I'll look at this again once the v2 API implementation is complete, which will be soon hopefully because I think v1 is unavailable as of Oct. 1/17.

**Comment by vipuldelwadia** [comment]
I've got a pull request open which changes TVRenamer to use the v2 API here; https://github.com/tvrenamer/tvrenamer/pull/240

You can check that out and create your own PR on top of that for the ordering work.

**Comment by eprenamer** [comment]
We definitely should move forward with the v2 API (and btw great job @vipuldelwadia !)  But for the record, v1 absolutely does have the DVD order information, and it is included in the 0.8 sources.

**Comment by trolley** [comment]
Interesting. I looked at the XML returned when I queried for Firefly and the DVD episode and season fields were empty.

**Comment by vipuldelwadia** [comment]
Interesting, because we definitely capture that information in tvrenamer currently: https://github.com/tvrenamer/tvrenamer/blob/master/src/main/org/tvrenamer/model/Episode.java#L26

I wonder if it's now broken in the XML API, following that big outage they had a little while ago?

**Comment by trolley** [comment]
I saw that. That's where I started, trying to create a separate list of DVD seasons but found it was always empty and eventually traced it back to looking at the XML returned by the query and finding there's no DVD info in it.

**Comment by vipuldelwadia** [comment]
I think at this point we should just switch to the new API

**Comment by eprenamer** [comment]
I agree, but I don't see the relevance to the DVD ordering.  Currently, when we look up a show, we download BOTH orderings.  And I think we should continue to do so with the new API.

It's a matter of what we do with that information after downloading, that is really the issue.  Get that right, and it won't matter which API we use (or even if we eventually move to a different provider, etc.)

BTW the sunset date has been postponed.

**Comment by eprenamer** [comment]
@trolley Not all shows have DVD information.  What show are you looking at?

One that I know is interesting is Robot Chicken, because unfortunately it has hard conflicts between the DVD ordering and the air ordering.

**Comment by trolley** [comment]
I'm using Firefly for my testing, but I know of a handful of shows I'm
using DVD ordering on:


   -

   Archer (2009)
   -

   Arrested Development
   -

   Buffy the Vampire Slayer
   -

   Firefly
   -

   Futurama
   -

   Robot Chicken
   -

   Seinfeld

I use Plex and set those series as DVD ordered.

**Comment by vipuldelwadia** [comment]
@trolley Firefly definitely has the DVD ordering information in the XML API -
 http://thetvdb.com/api/4A9560FF0B2670B2/series/78874/all/en.xml:

```xml
<Episode>
  <id>297989</id>
  <Combined_episodenumber>2.0</Combined_episodenumber>
  <Combined_season>1</Combined_season>
  <DVD_chapter></DVD_chapter>
  <DVD_discid></DVD_discid>
  <DVD_episodenumber>2.0</DVD_episodenumber>
  <DVD_season>1</DVD_season>
  <Director>|Joss Whedon|</Director>
  <EpImgFlag>2</EpImgFlag>
  <EpisodeName>The Train Job</EpisodeName>
  <EpisodeNumber>1</EpisodeNumber>
  <FirstAired>2002-09-20</FirstAired>
  <GuestStars>|Gregg Henry|Tom Towles|Andrew Bryniarski|Michael Fairman|Jeff Ricketts|Mark Sheppard|Valerie Red-Horse|Kevin Will|Lina Patel|David Reynolds|Dennis Cockrum|Eric Lange|Tawny Rene Hamilton|Michelle Ferrara|Drew Goddard|</GuestStars>
  <IMDB_ID>tt0579539</IMDB_ID>
  <Language>en</Language>
  <Overview>Serenity continues to travel through space conducting illegal runs across the star system, which is under the control of The Alliance. A powerful and fearsome mobster named Adelei Niska demands that the crew of Serenity pull a train robbery. The train is carrying medical supplies for a community, and Reynolds and Zoe get held on suspicion. They get free with Inara's help but realize they were tricked and try to return Niska's money. His henchman isn't too happy, but Mal &quot;persuades&quot; the next guy to take the money back. Meanwhile, two blue-handed, black-suited operatives close in on River.</Overview>
  <ProductionCode>1AGE01</ProductionCode>
  <Rating>8.1</Rating>
  <RatingCount>149</RatingCount>
  <SeasonNumber>1</SeasonNumber>
  <Writer>|Tim Minear|Joss Whedon|</Writer>
  <absolute_number>2</absolute_number>
  <filename>episodes/78874/297989.jpg</filename>
  <is_movie>0</is_movie>
  <lastupdated>1458047554</lastupdated>
  <seasonid>15791</seasonid>
  <seriesid>78874</seriesid>
  <thumb_added></thumb_added>
  <thumb_height>225</thumb_height>
  <thumb_width>400</thumb_width>
</Episode>
```

**Comment by trolley** [comment]
I’ll try again, thanks.
